scipy/linalg/_decomp_svd.py

```diff
@@ -299,6 +303,101 @@ def diagsvd(s, M, N):
         raise ValueError("Length of s must be M or N.")
 
 
+def higher_order_svd(
+    a: npt.ArrayLike, *, compact_rtol: float | None = None, check_finite: bool = True
```

Comment:
Nit: perhaps more consistent with the referenced paper to make this an upper case A?

```suggestion
    A: npt.ArrayLike, *, compact_rtol: float | None = None, check_finite: bool = True
```

Reply:
I used the same style as in the normal `svd` routine to keep it systematic.